packages/twenty-server/src/workspace/workspace-sync-metadata/workspace-sync-metadata.service.ts

```diff
@@ -76,20 +79,23 @@ export class WorkspaceSyncMetadataService {
         );
 
       // Save workspace migrations into the database
-      const workspaceMigrations = await workspaceMigrationRepository.save([
+      workspaceMigrations = await workspaceMigrationRepository.save([
         ...workspaceObjectMigrations,
         ...workspaceRelationMigrations,
       ]);
 
       // If we're running a dry run, rollback the transaction and do not execute migrations
-      if (options?.dryRun) {
+      if (!options.applyChanges) {
         this.logger.log('Running in dry run mode, rolling back transaction');
 
         await queryRunner.rollbackTransaction();
 
-        await this.workspaceLogsService.saveLogs(storage, workspaceMigrations);
```

Comment:
Move creation of logs file inside command, doesn't feel to be the correct place directly in the services